spacer/extract_features_utils.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+
+
+def gray2rgb(im):
+    """
+    Convert gray image to RGB image
+    :param im: gray image to be converted
+    :return: RGB image
+    """
+    w, h = im.shape
+    ret = np.empty((w, h, 3), dtype=np.uint8)
+    ret[:, :, 0] = im
+    ret[:, :, 1] = im
+    ret[:, :, 2] = im
+
+    return ret
+
+
+def crop_patch(im, rowcols, crop_size, scale, offsets=None):
```

Comment:
Also: maybe add type hints here? I realized I hadn't done so originally, but since this is now a shared util, seems worth it.

Reply:
... we really should have them for all methods. I just got lazy in the end.

Reply:
hah yea sure, I will go through all utils files and add type hints